src/main/java/com/direwolf20/buildinggadgets/common/building/implementation/Grid.java

```diff
@@ -0,0 +1,91 @@
+package com.direwolf20.buildinggadgets.common.building.implementation;
+
+import com.direwolf20.buildinggadgets.common.building.Region;
+import com.direwolf20.buildinggadgets.common.building.placement.IPlacementSequence;
+import com.google.common.collect.AbstractIterator;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+
+import javax.annotation.Nonnull;
+import java.util.Iterator;
+
+public final class Grid implements IPlacementSequence {
+
+    public static Grid create(BlockPos base, int range, int periodSize) {
+        return new Grid(base, range, periodSize);
+    }
+
+
+    private final int periodSize;
+    private final Region region;
+    private final BlockPos center;
+    private final int range;
+
+    protected Grid(BlockPos center, int range, int periodSize) {
+        this.region = Wall.clickedSide(center, EnumFacing.UP, range).getBoundingBox();
+        this.range = range;
+        this.center = center;
+        this.periodSize = periodSize;
+    }
+
+    /**
+     * For {@link #copy()}
+     */
+    private Grid(Region region, BlockPos center, int range, int periodSize) {
+        this.region = region;
+        this.center = center;
+        this.range = range;
+        this.periodSize = periodSize;
+    }
+
+    @Override
+    public Region getBoundingBox() {
+        return region;
+    }
+
+    @Override
+    public boolean contains(int x, int y, int z) {
```

Comment:
don't do that... This violates the contains contract... For cases like this I originally called it mayContain - if it is difficult to compute, one could just return true.